backend/controllers/authController.js

```diff
@@ -36,19 +45,31 @@ module.exports.register_post = async (req, res) => {
 module.exports.login_post = async (req, res) => {
   const { email, password } = req.body;
   try {
-    const user = await User.findOne({ email });
-    if (!user) {
+    const loggedInUser = await User.findOne({ email });
+    if (!loggedInUser) {
       return res.status(404).json({
         email: "Email not found",
       });
     }
-    const passwordMatch = await user.matchPassword(password);
+    const passwordMatch = await loggedInUser.matchPassword(password);
     if (passwordMatch) {
+      const user = {
+        userId: loggedInUser._id,
+          firstName: loggedInUser.firstName,
+          lastName: loggedInUser.lastName,
+          email: loggedInUser.email,
+      }
+      const response = {
+        data: {
+          user
+        }
+      }
+      const token = jwt.sign({ ...response }, process.env.JWT_SECRET, {
+        expiresIn: process.env.JWT_EXPIRES_IN,
+      });
```

Comment:
When creating the token, the data is put inside token. So now we have to configure the front end to decode this to get access to the data.